.travis.yml

```diff
@@ -1,31 +1,108 @@
-    - python: pypy3.5-6.0
-    - python: 2.7
-      env:
-          - PURE_PYTHON=1
-          - ZOPE_WATCH_CHECKERS=1
-    - python: 3.7
+
+    - name: "Python: 2.7, pure (no C extensions), with ZOPE_WATCH_CHECKERS enabled"
+      python: "2.7"
+      env: PURE_PYTHON=1 ZOPE_WATCH_CHECKERS=1
+
+    - name: "Python: 3.7, with ZOPE_WATCH_CHECKERS enabled"
+      python: 3.7
       env: ZOPE_WATCH_CHECKERS=1
-    - python: 3.7
+
+    - name: "Python: 3.7, pure (no C extensions)"
+      python: 3.7
       env: PURE_PYTHON=1
+
+    # manylinux wheel builds
+    - name: 64-bit manylinux wheels (all Pythons)
+      services: docker
+      env: DOCKER_IMAGE=quay.io/pypa/manylinux2010_x86_64
+      install: docker pull $DOCKER_IMAGE
+      script: bash .manylinux.sh
+
+    - name: 32-bit manylinux wheels (all Pythons)
+      services: docker
+      env: DOCKER_IMAGE=quay.io/pypa/manylinux2010_i686 PRE_CMD=linux32
+      install: docker pull $DOCKER_IMAGE
+      script: bash .manylinux.sh
+
+    # It's important to use 'macpython' builds to get the least
+    # restrictive wheel tag. It's also important to avoid
+    # 'homebrew 3' because it floats instead of being a specific version.
+    - name: Python 2.7 wheels for MacOS
+      os: osx
+      language: generic
+      env: TERRYFY_PYTHON='macpython 2.7.17'
+    - name: Python 3.5 wheels for MacOS
+      os: osx
+      language: generic
+      env: TERRYFY_PYTHON='macpython 3.5'
+    - name: Python 3.6 wheels for MacOS
+      os: osx
+      language: generic
+      env: TERRYFY_PYTHON='macpython 3.6.0'
+    - name: Python 3.7 wheels for MacOS
+      os: osx
+      language: generic
+      env: TERRYFY_PYTHON='macpython 3.7.0'
+
+before_install:
+  - |
+    if [[ "$TRAVIS_OS_NAME" == "osx" ]]; then
+      git clone https://github.com/MacPython/terryfy
+      source terryfy/travis_tools.sh
+      get_python_environment $TERRYFY_PYTHON venv
+    fi
+
 install:
-    - pip install -U pip setuptools
-    - pip install -U coveralls coverage
-    - pip install -U -e .[test,docs]
+  - pip install -U pip setuptools
+  - pip install -U coveralls coverage
+  # pip install zope.proxy here because because setup_requires will use
+  # easy_install, which fails on Python 3.5 on Mac OS
+  - pip install -U zope.proxy
+  - pip install -U -e .[test,docs]
+
 script:
-    - coverage run -m zope.testrunner --test-path=src
-    - coverage run -a -m sphinx -b doctest -d docs/_build/doctrees docs docs/_build/doctest
+  - python --version
+  - coverage run -m zope.testrunner --test-path=src
+  - coverage run -a -m sphinx -b doctest -d docs/_build/doctrees docs docs/_build/doctest
+  - python setup.py bdist_wheel
+
 after_success:
-    - coveralls
+  - coveralls
+  - |
+    if [[ $TRAVIS_TAG && "$TRAVIS_OS_NAME" == "osx" ]]; then
+      # macpython 3.5 doesn't support recent TLS protocols which causes twine
+      # upload to fail, so we use the system Python to run twine
+      /usr/bin/python -m pip install twine
```

Comment:
This needs a `/usr/bin/python -m ensurepip` first.

Reply:
Actually it needs much more, see https://github.com/zopefoundation/zope.interface/pull/148.  (I'd prefer to release zope.interface 1.7.1 to make sure that recipe works before copying it elsewhere.)